modules/optflow/src/variational_refinement.cpp

```diff
@@ -1102,8 +1102,8 @@ void VariationalRefinementImpl::calcUV(InputArray I0, InputArray I1, InputOutput
     int num_stripes = getNumThreads();
     Mat I0Mat = I0.getMat();
     Mat I1Mat = I1.getMat();
-    Mat &W_u = flow_u.getMatRef();
-    Mat &W_v = flow_v.getMatRef();
+    Mat W_u = flow_u.getMat();
+    Mat W_v = flow_v.getMat();
```

Comment:
I got error like below if add flow_u.assign(W_u) code at the end of function

" C++ exception with description "/home/pli1/opencv/modules/core/src/ocl.cpp:5036: error: (-215) u->refcount == 0 || u->tempUMat() in function upload" 

so I decided to keep the code "Mat &W_u = flow_u.getMatRef();" and pass a mat argument in "variational_refinement_processors[i]->calcUV(...)", please see latest update